include/subscriptions/wssSubscriber.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <spdlog/spdlog.h>
+
+#include <functional>
+#include <thread>
+#include <websocketpp/client.hpp>
+#include <websocketpp/config/asio_client.hpp>
+
+#include "mango_v3.hpp"
+
+typedef websocketpp::client<websocketpp::config::asio_tls_client> ws_client;
+typedef websocketpp::config::asio_client::message_type::ptr ws_message_ptr;
+typedef std::shared_ptr<boost::asio::ssl::context> context_ptr;
+
+namespace mango_v3 {
+namespace subscription {
+
+using json = nlohmann::json;
+
+class wssSubscriber {
+ public:
+  wssSubscriber(const std::string& account) : account(account) {}
+
+  ~wssSubscriber() {
+    if (runThread.joinable()) {
+      websocketpp::lib::error_code ec;
+      c.stop();
+      runThread.join();
+    }
+  }
+
+  void registerOnMessageCallback(
+      std::function<void(const json& message)> callback) {
+    onMessageCb = callback;
+  }
+
+  void registerOnCloseCallback(std::function<void()> callback) {
+    onCloseCb = callback;
+  }
+
+  void start() {
+    c.clear_access_channels(websocketpp::log::alevel::all);
+    c.init_asio();
+    c.set_tls_init_handler(
+        websocketpp::lib::bind(&wssSubscriber::on_tls_init, this));
+    c.set_open_handler(websocketpp::lib::bind(
+        &wssSubscriber::on_open, this, websocketpp::lib::placeholders::_1));
+    c.set_message_handler(websocketpp::lib::bind(
+        &wssSubscriber::on_message, this, websocketpp::lib::placeholders::_1,
+        websocketpp::lib::placeholders::_2));
+    c.set_close_handler(websocketpp::lib::bind(
+        &wssSubscriber::on_close, this, websocketpp::lib::placeholders::_1));
+
+    try {
+      websocketpp::lib::error_code ec;
+      ws_client::connection_ptr con = c.get_connection(MAINNET.endpoint, ec);

```

Comment:
`MAINNET.endpoint` should not be statically passed but we should create an endpoint argument in the constructor.

Reply:
again: